src/Ticket.php

```diff
@@ -4409,6 +4409,14 @@ public function showForm($ID, array $options = [])
             $options['_noupdate'] = true;
         }
 
+        // compute priority if is new ticket and user have no right to change priority
+        if ($this->isNewItem() && !$canpriority) {
+            $this->fields['priority'] = $this->computePriority(
+                $this->fields['urgency'] ?? 3,
+                $this->fields['impact'] ?? 3
+            );
+        }
+
```

Comment:
Maybe something like this?

```suggestion
        // compute priority if is new ticket and user have no right to change priority
        if ($this->isNewItem() && !$canpriority) {
            $this->fields['priority'] = $this->computePriority(
                $this->fields['urgency'] ?: $predefined_fields['urgency'] ?? 3 /* Medium */,
                $this->fields['impact'] ?: $predefined_fields['impact']  ?? 3 /* Medium */
            );
        }

```